tests/phpunit/tests/dependencies/scripts.php

```diff
@@ -586,6 +586,105 @@ public function test_get_normalized_script_args() {
 		$this->assertSame( $expected_args, $wp_scripts->get_data( 'footer-old', 'script_args' ) );
 	}
 
+	/**
+	 * Test script concatenation with deferred main script.
+	 *
+	 * @ticket 12009
+	 */
+	public function test_concatenate_with_defer_strategy() {
+		global $wp_scripts, $concatenate_scripts;
+
+		$old_value = $concatenate_scripts;
+		$concatenate_scripts = true;
+
+		$wp_scripts->do_concat    = true;
+		$wp_scripts->default_dirs = array( '/directory/' );
```

Comment:
Nitpick: we could make the default dir path a variable so it can easily be updated everywhere.